SU2_CFD/include/numerics/scalar/scalar_convection.hpp

```diff
@@ -130,9 +131,14 @@ class CUpwScalar : public CNumerics {
         q_ij += 0.5 * (V_i[iDim + idx.Velocity()] + V_j[iDim + idx.Velocity()]) * Normal[iDim];
       }
     }
-
-    a0 = 0.5 * (q_ij + fabs(q_ij));
-    a1 = 0.5 * (q_ij - fabs(q_ij));
+    
+    if(config->GetKind_Upwind_Species() == UPWIND::BOUNDED_SCALAR){
+      a0 = max(0.0, MassFlux);
+      a1 = min(0.0, MassFlux);
```

Comment:
I think the simplest solution to what we talked about in the dev meeting is to divide here by density, and then allow the derived classes to re-introduce density if they require it.
```suggestion
      a0 = max(0.0, MassFlux) / Density_i;
      a1 = min(0.0, MassFlux) / Density_j;
```